lakesoul-common/src/main/java/com/dmetasoul/lakesoul/meta/jnr/NativeMetadataJavaClient.java

```diff
@@ -234,43 +231,67 @@ public JniWrapper executeQuery(Integer queryType, List<String> params) {
             int retryCounter = NATIVE_METADATA_MAX_RETRY_ATTEMPTS;
             while (retryCounter >= 0) {
                 try {
-                    final CompletableFuture<Integer> future = new CompletableFuture<>();
-                    getLibLakeSoulMetaData().execute_query(
+                    final CompletableFuture<Integer> queryFuture = new CompletableFuture<>();
+                    Pointer queryResult = getLibLakeSoulMetaData().execute_query(
                             new ReferencedIntegerCallback((result, msg) -> {
                                 if (msg.isEmpty()) {
-                                    future.complete(result);
+                                    queryFuture.complete(result);
                                 } else {
-                                    future.completeExceptionally(new SQLException(msg));
+                                    queryFuture.completeExceptionally(new SQLException(msg));
                                 }
                             }, getIntegerCallbackObjectReferenceManager()),
                             tokioRuntime,
                             tokioPostgresClient,
                             preparedStatement,
                             queryType,
-                            String.join(PARAM_DELIM, params),
-                            queryType < DAO_TYPE_QUERY_LIST_OFFSET ? sharedBuffer.address() : largeSharedBuffer.address()
+                            String.join(PARAM_DELIM, params)
+//                            queryType < DAO_TYPE_QUERY_LIST_OFFSET ? sharedBuffer.address() : largeSharedBuffer.address()
                     );
-                    Integer len = future.get(timeout, TimeUnit.MILLISECONDS);
+                    Integer len = queryFuture.get(timeout, TimeUnit.MILLISECONDS);
                     if (len < 0) return null;
```

Comment:
Should throw an exception when encountering error

Reply:
It's already wrapped by try-catch clause.